### PR TITLE
Add filter and comparator props to DeviceTable and AccessCodeTable

### PR DESCRIPTION
**Pull request by razor-x**

### CODE REVIEW

File: src/lib/seam/components/AccessCodeTable/AccessCodeTable.tsx [hunk 1]
```diff
@@ -27,23 +27,53 @@ import { Title } from 'lib/ui/typography/Title.js'
...
   className?: string
 }
 
+type AccessCode = UseAccessCodesData[number]
+
+const defaultAccessCodeFilter = (
+  accessCode: AccessCode,
+  searchInputValue: string
+) => {
+  if (searchInputValue === '') return true
```
Comment (xplato) [inline src/lib/seam/components/AccessCodeTable/AccessCodeTable.tsx:48]: 🐠 Pretty minor, but we could add a `.trim()` here unless there's a reason someone would search for an empty space(s)

File: src/lib/seam/components/AccessCodeTable/AccessCodeTable.tsx [hunk 2]
```diff
@@ -27,23 +27,53 @@ import { Title } from 'lib/ui/typography/Title.js'
...
+  return new RegExp(searchInputValue, 'i').test(accessCode.name ?? '')
+}
+
 export function AccessCodeTable({
   deviceId,
   onBack,
+  accessCodeFilter = defaultAccessCodeFilter,
```
Comment (xplato) [inline src/lib/seam/components/AccessCodeTable/AccessCodeTable.tsx:55]: Nice, good idea! Out of curiosity, was this a requested feature?

**Review by xplato — APPROVED** [review]
Awesome; good consumer-friendly pattern 👌

File: src/lib/seam/components/AccessCodeTable/AccessCodeTable.tsx (hunk 2 above)
Comment (razor-x) [reply src/lib/seam/components/AccessCodeTable/AccessCodeTable.tsx:55]: Yup :)

**Pull request merged**